ArcGISRuntimeSDKQt_CppSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.cpp

```diff
@@ -0,0 +1,304 @@
+
+    if (networkSource->sourceType() == UtilityNetworkSourceType::Junction)
+    {
+      const QString assetGroupFieldName = dynamic_cast<ArcGISFeatureTable*>(m_feature->featureTable())->subtypeField();
+      const int assetGroupCode = m_feature->attributes()->attributeValue(assetGroupFieldName).toInt();
+      const UtilityAssetGroup* assetGroup = nullptr;
+
+      for (const UtilityAssetGroup* group : networkSource->assetGroups())
+      {
+        if (group->code() == assetGroupCode)
+        {
+          assetGroup = group;
+          break;
+        }
+      }
+
+      const int assetTypeCode = m_feature->attributes()->attributeValue("assettype").toInt();
+
+      const UtilityAssetType* assetType = nullptr;
+      for (const UtilityAssetType* type : assetGroup->assetTypes())
+      {
+        if (type->code() == assetTypeCode)
+        {
+          assetType = type;
+          break;
+        }
+      }
+
+      m_terminals = assetType->terminalConfiguration()->terminals();
+
+      if (m_terminals.size() > 1)
+      {
+        terminalDialogVisisble = true;
+        emit terminalDialogVisisbleChanged();
+        return;
+      }
+      else if (m_terminals.size() == 1)
+      {
+        element = m_utilityNetwork->createElementWithArcGISFeature(m_feature, m_terminals[0]);
+      }
+      else
+      {
+        return;
+      }
+
+    }
+    else if (networkSource->sourceType() == UtilityNetworkSourceType::Edge)
+    {
+      element = m_utilityNetwork->createElementWithArcGISFeature(m_feature, nullptr ,this);
+
+      // Compute how far tapped location is along the edge feature.
+      if (m_feature->geometry().geometryType() == GeometryType::Polyline)
+      {
+        const Polyline line = GeometryEngine::removeZ(m_feature->geometry());
+        // Set how far the element is along the edge.
+        element->setFractionAlongEdge(GeometryEngine::fractionAlong(line, m_clickPoint, -1));
+      }
+    }
+    else
+    {
+      return;
+    }
+
+    updateTraceParams(element);
+  });
+}
+
+FindFeaturesUtilityNetwork::~FindFeaturesUtilityNetwork() = default;
+
+void FindFeaturesUtilityNetwork::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<FindFeaturesUtilityNetwork>("Esri.Samples", 1, 0, "FindFeaturesUtilityNetworkSample");
+}
+
+MapQuickView* FindFeaturesUtilityNetwork::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void FindFeaturesUtilityNetwork::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void FindFeaturesUtilityNetwork::multiTerminalIndex(const int &index)
+{
+  if (m_terminals.isEmpty())
+    return;
+  if (!m_feature)
+    return;
+
+  UtilityElement* element;
```

Comment:
since the only place this variable will be assigned is the next line, this should just all be on one line - `UtilityElement* element = m_utilityNetwork->createElementWithArcGISFeature(m_feature, m_terminals[index]);`